src/crimson/osd/heartbeat.cc

```diff
@@ -310,19 +310,21 @@ seastar::future<> Heartbeat::maybe_share_osdmap(
   }
   auto& peer = found->second;
 
-  if (m->map_epoch > peer.get_projected_epoch()) {
-    logger().debug("{} updating peer {} session's projected_epoch"
-                   "from {} to ping map epoch of {}",
-                   __func__, from, peer.get_projected_epoch(),
-                   m->map_epoch);
-    peer.set_projected_epoch(m->map_epoch);
+  // Peer already has newer projeceted osdmap epoch,
+  // no new incremental maps to share.
+  if (current_osdmap_epoch <= peer.get_projected_epoch()) {
+    logger().debug("{} our osdmap epoch of {} is not newer "
+                   " than peer {} projected_epoch of {}",
+                   __func__, current_osdmap_epoch, from,
+                   peer.get_projected_epoch());
+    return seastar::now();
   }
 
-  if (current_osdmap_epoch <= peer.get_projected_epoch()) {
-    logger().debug("{} peer {} projected_epoch {} is already later "
-		   "than our osdmap epoch of {}",
-		   __func__ , from, peer.get_projected_epoch(),
-		   current_osdmap_epoch);
+  if (m->map_epoch <= peer.get_projected_epoch()) {
```

Comment:
This comparison only really tells us that we already have an outstanding message, it's neither a necessary nor a sufficient condition to determine whether we need to send a map.